main.js

```diff
@@ -702,12 +677,51 @@ function main() {
                 [mathLines[i].end.x, mathLines[i].end.y]
             ])
             if (interSectPt) {
-                intersections.push(interSectPt);
+                intersections.push([round(interSectPt[0]), round(interSectPt[1])]);
```

Comment:
Any particular reason for doing the rounding here rather than at display time?

Reply:
Number imprecision makes the numbers off by 1e-8 sometimes, so the line is not actually colliding with the other lines at its ends. Rounding the intersections makes sure that we get the intersections and new vertices at the right places.

Rounding is also done somewhere else, so may be unnecessary here.